doc/sphinx/grammar.rst

```diff
@@ -0,0 +1,2247 @@
+                   : `module_expr` `functor_app_annot`
+   module_type_inl : ! `module_type`
+                   : `module_type` `functor_app_annot`
+   module_binder : ( `export_token` `ident_list` : `module_type_inl` )
+   module_expr : `module_expr_atom`
+               : `module_expr` `module_expr_atom`
+   module_expr_atom : `qualid`
+                    : ( `module_expr` )
+   with_declaration : Definition `fullyqualid` `univ_decl_opt2` := `lconstr`
+                    : Module `fullyqualid` := `qualid`
+   module_type : `qualid`
+               : ( `module_type` )
+               : `module_type` `module_expr_atom`
+               : `module_type` with `with_declaration`
+   section_subset_expr : `starredidentref_list_opt`
+                       : `ssexpr35`
+   starredidentref_list_opt : `starredidentref_list_opt` `starredidentref`
+                            : `empty`
+   starredidentref : `ident`
+                   : `ident` *
+                   : Type
+                   : Type *
+   ssexpr35 : - `ssexpr50`
+            : `ssexpr50`
+   ssexpr50 : `ssexpr0` - `ssexpr0`
+            : `ssexpr0` + `ssexpr0`
+            : `ssexpr0`
+   ssexpr0 : `starredidentref`
+           : ( `starredidentref_list_opt` )
+           : ( `starredidentref_list_opt` ) *
+           : ( `ssexpr35` )
+           : ( `ssexpr35` ) *
+   arguments_modifier : simpl nomatch
+                      : simpl never
+                      : default implicits
+                      : clear implicits
+                      : clear scopes
+                      : rename
+                      : assert
+                      : extra scopes
+                      : clear scopes and implicits
+                      : clear implicits and scopes
+   scope : % `IDENT`
+   argument_spec : `exclam_opt` `name` `scope_opt`
+   exclam_opt : !
+              : `empty`
+   scope_opt : `scope`
+             : `empty`
+   argument_spec_block : `argument_spec`
+                       : /
+                       : ( `argument_spec_list` ) `scope_opt`
+                       : [ `argument_spec_list` ] `scope_opt`
+                       : { `argument_spec_list` } `scope_opt`
+   argument_spec_list : `argument_spec_list` `argument_spec`
+                      : `argument_spec`
+   more_implicits_block : `name`
+                        : [ `name_list` ]
+                        : { `name_list` }
+   strategy_level : expand
+                  : opaque
+                  : `INT`
+                  : - `INT`
+                  : transparent
+   instance_name : `ident_decl` `binders`
+                 : `empty`
+   hint_info : | `natural_opt2` `constr_pattern_opt`
+             : `empty`
+   reserv_list : `reserv_tuple_list`
+               : `simple_reserv`
+   reserv_tuple_list : `reserv_tuple_list` `reserv_tuple`
+                     : `reserv_tuple`
+   reserv_tuple : ( `simple_reserv` )
+   simple_reserv : `ident_list` : `lconstr`
+   command : Comments `comment_list_opt`
```

Comment:
It is my impression that `command` is typical of something that we don't actually want to see appear in the grammar of the reference manual and prefer to be documented only using the `.. cmd::` or `.. cmdv::` directives.
Will it be possible to do this kind of mapping? [OK, this is documented later that this is not yet supported but is planned.]

Reply:
My best idea so far was to pick the longest character-by-character match of the text after `.. tacn::` with `simple_tactic` production rather than requiring an additional comment line above each tacn.  The results were pretty strange because the doc and the grammar really in sync.  You'd want to fix all of them simultaneously.

This approach is probably workable if the `tacn` gives the full syntax for the tactic (no omissions).

Not sure what could be done for the `tacv`s.  Seems like it would be hard to indicate what parts of the syntax to include without being totally tedious--or giving up on the automatic updating.